### ISSUE TITLE
Bug in Zeus Wallet v0.9.0: Payment Options Not Working as Expected

### ISSUE THREAD

**Issue by Perlover** [ZeusLN/zeus#2403]
**Description:**

In Zeus Wallet 0.9.0, there appears to be a bug when sending a payment and customizing the individual settings of the payment.

Steps to reproduce:
1. Attempt to send a payment.
2. Click on the individual settings for the payment.
3. Options appear, such as enabling or disabling multipath payments, and specifying the number of shards to send.
4. I tried setting the number of shards to 1 or disabling the multipath payment option.
5. I also specified an outgoing channel for the payment.

Expected behavior:
- The payment should be sent in one part when the multipath payment is disabled, or according to the selected number of shards.

Actual behavior:
- Despite disabling the multipath payment option or setting the number of shards to 1, the payment still splits into multiple parts. It appears that these options are not functioning as intended.

P.S. I would like to clarify that to check how many payments are being sent, I exited to the main screen while the payment was in progress and then went to the specific channel list. There, I could see that HTLC payments were pending in multiple channels.

Thank you for reviewing this issue!

**Comment by kaloudis** [comment]
@Perlover could you provide your backend and corresponding version number?

**Comment by Perlover** [comment]
This refers to the Zeus wallet, which has an internal LND server, and it's version 0.9.0 of Zeus itself.

**Comment by kaloudis** [comment]
@Perlover what version of LND and what interface are using to access it (REST, LNC)?

**Comment by Perlover** [comment]
Let me reiterate once again. The Zeus wallet is not being used as an interface to an external LND server elsewhere. The Zeus wallet itself has its own internal LND server. How it communicates is a question for the developers of the Zeus wallet.

**Comment by Perlover** [comment]
When I go into the settings and check the information about the internal LND server built into the Zeus wallet, I see version 0.18.2. I hope this information helps you, although I don’t think there could be a different LND version in Zeus wallet 0.9.0 other than the one specified by the developers.

**Comment by rpallai** [comment]
FYI: This also happens with external LND versions 0.18.0 and 0.18.3 connected via REST — it always attempts to use AMP, even when it's disabled.

**Comment by remcoros** [comment]
I noticed something similar with setting a max fee. The payment went through with a much higher fee than the custom maximum I've set. On Zeus 0.8 it would fail with a payment route error (which was expected). On 0.9 the payment was successful, but ignored the max fee. Both used external LND 0.18 (REST).

**Comment by kaloudis** [comment]
@Perlover @rpallai @remcoros have you guys been able to test against ZEUS v0.9.1-beta?

**Comment by remcoros** [comment]
> @Perlover @rpallai @remcoros have you guys been able to test against ZEUS v0.9.1-beta?

Yes, still not working for me. This was with a rebalancing transaction.

- Receive: create new invoice of 100k sats
- Send: paste invoice
- Set absolute max fee to 1 sat
- Set first hop and last hop (to rebalance)
- Send transaction

Transaction succeeded with a cost of 67 sats, ignoring the max fee of 1 sat.

As far as I can remember, with Zeus 0.8 it would fail with a routing error, until I raised the max fee higher. Let me know if you need more details, I took screenshots and can send them on telegram.

**Comment by remcoros** [comment]
> Still on LND REST v0.18.0? Any chance you're on Umbrel?

LND 0.18.3 on StartOS. I'll do a test with Zeus 0.8 to see if it's maybe LND.

**Comment by remcoros** [comment]
> > Still on LND REST v0.18.0? Any chance you're on Umbrel?
> 
> LND 0.18.3 on StartOS. I'll do a test with Zeus 0.8 to see if it's maybe LND.

Definitely an issue since Zeus 0.9.0, with 0.8.5-hotfix the payment fails immediately with a route error. Updated to 0.9.1-rc2 and it goes through fine with a higher fee.

**Comment by shubhamkmr04** [comment]
We have addressed the bug fix here https://github.com/ZeusLN/zeus/pull/2459. It will be out in our next v0.9.1-beta3

**Comment by kaloudis** [comment]
@remcoros @Perlover @rpallai The fix is now available in the latest test build. Hope you check it out:

https://github.com/ZeusLN/zeus/releases/tag/v0.9.1-rc1

**Comment by remcoros** [comment]
Confirmed working for me on 0.9.1-rc2 
thanks @shubhamkmr04 and @kaloudis !

**Comment by Perlover** [comment]
> @remcoros @Perlover @rpallai The fix is now available in the latest test build. Hope you check it out:
> 
> https://github.com/ZeusLN/zeus/releases/tag/v0.9.1-rc1

Hello,

I tried updating, but the version I have on my phone is from Google Play and it's signed with a key from that store. The version you're providing seems to be signed with a different key, so I'm unable to update directly. To test your version, I will need to install it on another device. I will try to do that within the next couple of days. Thank you for your understanding!

**Comment by Perlover** [comment]
> @remcoros @Perlover @rpallai The fix is now available in the latest test build. Hope you check it out:
> 
> https://github.com/ZeusLN/zeus/releases/tag/v0.9.1-rc1

Hello,
I have tested version 0.9.1 RC2. I disabled the MPP option, and the payment failed when I didn’t have enough funds in a single channel. However, when I enabled the MPP option, the payment succeeded and was processed across two channels. Therefore, the MPP option is correctly managing payments when enabled.

By the way, I’d like to suggest a potential improvement. The default settings in the wallet — 5% fee limit and 1000 satoshi maximum — often result in failed payments, especially for larger amounts. It might be more practical to set a default of either 5000 satoshi or a 1% fee limit. Another possible approach could be similar to the SBW wallet, which used both a maximum percentage fee and a "tolerable fee." In that case, even if the percentage exceeded the limit for smaller payments, the transaction could still go through as long as the fee remained under the tolerable absolute satoshi amount.
